src/database/sqlite.rs

```diff
@@ -43,6 +43,10 @@ static MIGRATIONS: &[&str] = &[
     "INSERT INTO transaction_details SELECT txid, timestamp, received, sent, fee, height FROM transaction_details_old;",
     "DROP TABLE transaction_details_old;",
     "ALTER TABLE utxos ADD COLUMN is_spent;",
+    "ALTER TABLE utxos RENAME TO utxos_old;",
+    "CREATE TABLE utxos (value INTEGER, keychain TEXT, vout INTEGER, txid BLOB, script BLOB, is_spent BOOLEAN DEFAULT 0);",
+    "INSERT INTO utxos SELECT value, keychain, vout, txid, script, is_spent FROM utxos_old;",
+    "DROP TABLE utxos_old;",
```

Comment:
This should be at the end of the migrations. 

See: https://github.com/bitcoindevkit/bdk/blob/master/src/database/sqlite.rs#L953-L975

We apply all the migrations we have, and then we save the index of the migration array of the last migration we applied. For this reason, the migration array must be append-only: otherwise clients when migrating might re-do steps they've already done!

If we kept the code like that, a client which is in master at the latest migration (`"CREATE UNIQUE INDEX idx_utxos_txid_vout ON utxos(txid, vout);"`) would re-do the last 4 migrations of the array (from `"DELETE FROM transaction_details;"` to `"CREATE UNIQUE INDEX idx_utxos_txid_vout ON utxos(txid, vout);"`), instead of applying the migrations you just added. Does that make sense?

For the same reason, you shouldn't delete the `"DROP INDEX idx_txid_vout;"` line.

Reply:
Thanks @danielabrozzoni! That makes sense.